### PR TITLE
prepare for Julia v1.0

### PR DESCRIPTION
**Pull request by RalphAS**
This attempts to run benchmarks in a useful way, and updates CI.

Edit: forcing commit updates for trivial changes to Travis script (I initially copied from a dubious source.)

### CODE REVIEW

**Comment by RalphAS** [comment]
FWIW, here is what I get for benchmark failures on v1.0 (since they don't run on CI):
```julia
[ Info: running benchmarks
┌ Warning: ColorView: failed on Main.ImageCoreTests.mysum_elt_boundscheck, time ratio 10.934298118668597, tol 5
└ @ Main.ImageCoreTests ~/.julia/dev/ImageCore/test/benchmarks.jl:110
┌ Warning: ChannelView: failed on Main.ImageCoreTests.mysum_index_boundscheck, time ratio 10.949593069046994, tol 10
└ @ Main.ImageCoreTests ~/.julia/dev/ImageCore/test/benchmarks.jl:103
┌ Warning: ColorView: failed on Main.ImageCoreTests.mysum_index_boundscheck, time ratio 11.205204240492254, tol 5
└ @ Main.ImageCoreTests ~/.julia/dev/ImageCore/test/benchmarks.jl:110
┌ Warning: ChannelView: failed on Main.ImageCoreTests.mysum_elt_inbounds, time ratio 11.179844778001025, tol 10
└ @ Main.ImageCoreTests ~/.julia/dev/ImageCore/test/benchmarks.jl:103
┌ Warning: ColorView: failed on Main.ImageCoreTests.mysum_elt_inbounds, time ratio 11.669422126081582, tol 3
└ @ Main.ImageCoreTests ~/.julia/dev/ImageCore/test/benchmarks.jl:110
┌ Warning: ColorView: failed on Main.ImageCoreTests.mysum_index_inbounds_simd, time ratio 12.153904358353511, tol 3
└ @ Main.ImageCoreTests ~/.julia/dev/ImageCore/test/benchmarks.jl:110
┌ Warning: ColorView: failed on Main.ImageCoreTests.myfill1!, time ratio 7.695395637234977, tol 3
└ @ Main.ImageCoreTests ~/.julia/dev/ImageCore/test/benchmarks.jl:110
```

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+#         - julia -e 'import Pkg; Pkg.clone(pwd()); Pkg.build("ImageCore")'
+#         - julia -e 'import Pkg; Pkg.add("Documenter")'
+#         - julia -e 'import ImageCore; include("docs/make.jl")'
 after_success:
-  # push coverage results to Codecov
-  - julia -e 'cd(Pkg.dir("ImageCore")); Pkg.add("Coverage"); using Coverage; Codecov.submit(Codecov.process_folder())'
-  # update the documentation
-  - julia -e 'Pkg.add("Documenter")'
-  - julia -e 'cd(Pkg.dir("ImageCore")); ENV["DOCUMENTER_DEBUG"] = "true"; include(joinpath("docs", "make.jl"))'
+    - julia -e 'using Pkg; cd(Pkg.dir("ImageCore")); Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'
```
Comment (Evizero) [inline .travis.yml]: `Pkg.dir` is deprecated in 1.0

File: test/benchmarks.jl [hunk 2]
```diff
@@ -88,25 +88,29 @@ chanvtol = Dict(mysum_index_inbounds_simd => 20,   # @simd doesn't work for Chan
...
     for f in suite
         # Channelview
         t_ar, t_cv = testf(f, a, vchan, 10^2)
         tol = haskey(chanvtol, f) ? chanvtol[f] : chanvdefault
         if t_cv >= tol*t_ar
-            println("ChannelView: failed on $f, time ratio $(t_cv/t_ar), tol $tol")
+            @warn "ChannelView: failed on $f, time ratio $(t_cv/t_ar), tol $tol"
         end
-        @test t_cv < tol*t_ar
+        # @test t_cv < tol*t_ar
```
Comment (Evizero) [inline test/benchmarks.jl:106]: maybe `@test_broken` instead?

File: test/benchmarks.jl [hunk 3]
```diff
@@ -58,7 +58,7 @@ function test_getindex(f, ar, cv, n)
     f_cv = Ref(f(cv))
     for i = 1:n
         t_ar[i] = (tstart = time(); f_ar[] = f(ar); time()-tstart)
-        t_ar[i] = (tstart = time(); f_cv[] = f(cv); time()-tstart)
+        t_cv[i] = (tstart = time(); f_cv[] = f(cv); time()-tstart)
```
Comment (Evizero) [inline test/benchmarks.jl:61]: good catch!

File: .travis.yml (hunk 1 above)
Comment (RalphAS) [reply .travis.yml]: Ok, there's apparently no need for for the `cd` now anyway.

File: test/benchmarks.jl (hunk 2 above)
Comment (RalphAS) [reply test/benchmarks.jl:106]: Some of the comparisons work and others don't so I figured warnings would be more useful now.

**Comment by timholy** [comment]
I get similar failures. FYI Keno told me he'll keep working on
reinterpretarray performance, maybe we'll get it in 1.0.x or 1.1. It's
unfortunate in the short term but I am confident it will work out in the
long run, because using reinterpretarray is a much cleaner solution.

I am in transit today. Interpolations needs a ton of work and I will be
plugging away at it. It seems we need to do a breaking release (gradient)
and we might as well break everything at once. Then I hope to get back to
image packages.

File: .travis.yml [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
+    email: false
+script:
+    - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
+    - julia --check-bounds=yes --color=yes -e 'import Pkg; Pkg.clone(pwd()); Pkg.build("ImageCore"); Pkg.test("ImageCore"; coverage=true)';
+
+# control of actual deployment is in the Documenter make script
+jobs:
+  include:
+    - stage: deploy
+      julia: 0.7
```
Comment (timholy) [inline .travis.yml:19]: We'll want to change this to 1.0 sometime, but this is fine for now.

File: test/benchmarks.jl (hunk 2 above)
Comment (timholy) [reply test/benchmarks.jl:106]: Could also use `@test_skip` but frankly I think us seeing the status may be more important than automated testing here.

**Review by timholy — APPROVED** [review]
Nice work as usual! Merging.

**Pull request merged**